gravatar/src/main/java/com/gravatar/services/ProfileService.kt

```diff
@@ -22,95 +15,34 @@ public class ProfileService(private val okHttpClient: OkHttpClient? = null) {
-                            }
-                        } else {
-                            // Log the response body for debugging purposes if the response is not successful
-                            Logger.w(
-                                LOG_TAG,
-                                "Network call unsuccessful trying to get Gravatar profile: $response.body",
-                            )
-                            getProfileListener.onError(errorTypeFromHttpCode(response.code()))
-                        }
-                    }
-                }
-
-                override fun onFailure(call: Call<Profile>, t: Throwable) {
-                    coroutineScope.launch {
-                        getProfileListener.onError(t.error())
-                    }
-                }
-            },
-        )
-    }
-
-    /**
-     * Fetches a Gravatar profile for the given email address.
-     *
-     * @param email The email address to fetch the profile for
-     * @param getProfileListener The listener to notify of the fetch result
-     */
-    public fun fetchWithListener(email: Email, getProfileListener: GravatarListener<Profile, ErrorType>) {
-        fetchWithListener(email.hash(), getProfileListener = getProfileListener)
-    }
-
-    /**
-     * Fetches a Gravatar profile for the given hash.
-     *
-     * @param hash The hash to fetch the profile for
-     * @param getProfileListener The listener to notify of the fetch result
-     */
-    public fun fetchWithListener(hash: Hash, getProfileListener: GravatarListener<Profile, ErrorType>) {
-        fetchWithListener(hash.toString(), getProfileListener = getProfileListener)
-    }
-
-    /**
-     * Fetches a Gravatar profile for the given username.
-     *
-     * @param username The username to fetch the profile for
-     * @param getProfileListener The listener to notify of the fetch result
-     */
-    public fun fetchWithListenerByUsername(
-        username: String,
-        getProfileListener: GravatarListener<Profile, ErrorType>,
-    ) {
-        fetchWithListener(username, getProfileListener = getProfileListener)
-    }
-
     /**
      * Fetches a Gravatar profile for the given hash or username.
      *
      * @param hashOrUsername The hash or username to fetch the profile for
      * @return The fetched profile
      */
     public suspend fun fetch(hashOrUsername: String): Result<Profile, ErrorType> {
-        return suspendCoroutine {
-            fetchWithListener(
-                hashOrUsername,
-                object : GravatarListener<Profile, ErrorType> {
-                    override fun onSuccess(response: Profile) {
-                        it.resume(Result.Success(response))
-                    }
-
-                    override fun onError(errorType: ErrorType) {
-                        it.resume(Result.Failure(errorType))
-                    }
-                },
-            )
+        val service = GravatarSdkDI.getGravatarApiV3Service(okHttpClient)
```

Comment:
Also this `service` local variable can be defined as a property to avoid recreating the retrofit service like commented above